crates/router/src/core/user.rs

```diff
@@ -1234,6 +1266,95 @@ pub async fn create_internal_user(
     Ok(ApplicationResponse::StatusOk)
 }
 
+pub async fn create_tenant_user(
+    state: SessionState,
+    request: user_api::CreateTenantRequest,
+) -> UserResponse<()> {
+    let key_manager_state = &(&state).into();
+
+    let (merchant_id, _) = state
+        .store
+        .list_merchant_and_org_ids(key_manager_state, 1, 0)
+        .await
+        .change_context(UserErrors::InternalServerError)
+        .attach_printable("Failed to get merchants list for org")?
+        .pop()
+        .ok_or(UserErrors::InternalServerError)
+        .attach_printable("No merchants found in the tenancy")?;
+
+    let key_store = state
+        .store
+        .get_merchant_key_store_by_merchant_id(
+            key_manager_state,
+            &merchant_id,
+            &state.store.get_master_key().to_vec().into(),
+        )
+        .await
+        .change_context(UserErrors::InternalServerError)
+        .attach_printable("Error while fetching the key store by merchant_id")?;
+
+    let merchant_account = state
+        .store
+        .find_merchant_account_by_merchant_id(key_manager_state, &merchant_id, &key_store)
+        .await
+        .change_context(UserErrors::InternalServerError)
+        .attach_printable("Error while fetching the merchant_account by merchant_id")?;
```

Comment:
this can be skipped is below function only requires merchant id and org id and not the whole merchant account.